copy/copy.go

```diff
@@ -1234,6 +1242,10 @@ func (c *copier) copyBlobFromStream(ctx context.Context, srcStream io.Reader, sr
 	isCompressed := decompressor != nil
 	destStream = bar.ProxyReader(destStream)
 
+	if expectedCompressionFormat, known := expectedCompressionFormats[srcInfo.MediaType]; known && compressionFormat.Name() != expectedCompressionFormat.Name() {
```

Comment:
The `compressionFormat` value is invalid if `!isCompressed` (the `Name` call won’t crash given the current implementation, but calling `CompressStream` will); so `!isCompressed` needs to be checked first.

(And it turns out `DetectCompressionFormat` is under-documented.)

Reply:
Added a check for `isCompressed` before calling at `compressionFormat.Name()`, and tweaked the godoc for `DetectCompressionFormat` to describe the returned Algorithm value.